### PR TITLE
feat: add Versions

### PR DESCRIPTION
**Pull request by artemmufazalov**
Add versions info to Cluster page

Versions previously was present only internally, so I highlighted changes, that I made to fit open-source ui

### CODE REVIEW

File: src/containers/Cluster/Cluster.tsx [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-import {useRouteMatch} from 'react-router';
-import {ClusterInfo} from '../ClusterInfo/ClusterInfo';
 
 const b = cn('cluster');
 
 interface ClusterProps {
     additionalClusterInfo?: any;
     additionalTenantsInfo?: any;
     additionalNodesInfo?: any;
+    additionalVersionsProps?: AdditionalVersionsProps;
```
Comment (artemmufazalov) [inline src/containers/Cluster/Cluster.tsx:20]: The naming is intentional, I don't like `...info`. I want to alter it for other `additional...` as well. However, I'm not sure, that it is the best, so if you have better ideas, you are welcome

File: src/containers/ClusterInfo/ClusterInfo.tsx [hunk 2]
```diff
@@ -59,10 +75,22 @@ export const ClusterInfo = ({clusterTitle, additionalClusterInfo = []}: ClusterI
...
     const fetchData = useCallback(() => {
         dispatch(getClusterInfo(clusterName ? String(clusterName) : undefined));
+        dispatch(getClusterNodes());
     }, [dispatch, clusterName]);
 
     useAutofetcher(fetchData, [fetchData], true);
 
+    const versionToColor = useMemo(() => {
+        if (additionalVersionsProps?.getVersionToColorMap) {
+            return additionalVersionsProps?.getVersionToColorMap();
```
Comment (artemmufazalov) [inline src/containers/ClusterInfo/ClusterInfo.tsx:85]: In internal version we have constant versions to color mapping, so colours shouldn't be calculated

File: src/containers/ClusterInfo/ClusterInfo.tsx [hunk 3]
```diff
@@ -59,10 +75,22 @@ export const ClusterInfo = ({clusterTitle, additionalClusterInfo = []}: ClusterI
...
+        }
+        return parseVersionsToVersionToColorMap(cluster?.Versions);
+    }, [additionalVersionsProps, cluster]);
+
+    const versionsValues = useMemo(() => {
+        return parseNodesToVersionsValues(nodes, versionToColor);
```
Comment (artemmufazalov) [inline src/containers/ClusterInfo/ClusterInfo.tsx:91]: As we don't have nodes count for specific versions in `viewer/json/cluster` yet, we calculate it

File: src/containers/ClusterInfo/ClusterInfo.tsx [hunk 4]
```diff
@@ -128,39 +156,56 @@ export const ClusterInfo = ({clusterTitle, additionalClusterInfo = []}: ClusterI
         return info;
     };
 
-    if (loading && !wasLoaded) {
+    if ((loading && !wasLoaded) || (nodesLoading && !nodesWasLoaded)) {
```
Comment (artemmufazalov) [inline src/containers/ClusterInfo/ClusterInfo.tsx:159]: I didn't merge `clusterNodes` into `cluster` state, as we plan to change the navigation and `Versions` will be a separate tab. `clusterNodes` will be requested in `Cluster`, loading state will be different for `ClusterInfo` and `Versions`

File: src/containers/Versions/GroupedNodesTree/GroupedNodesTree.tsx
Comment (artemmufazalov) [inline src/containers/Versions/GroupedNodesTree/GroupedNodesTree.tsx:1]: Changes from internal version:
- Add `additionalVersionsProps`
- calculate nodes `uptime` in reducer, use `PreparedClusterNode`

File: src/containers/Versions/NodesTable/NodesTable.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                row.Endpoints && row.Endpoints.find((item) => item.Name === 'http-mon')?.Address;
+            const host = row.Host && `${row.Host}${port || ''}`;
+            const title = host || 'unknown';
+
+            const nodeHref =
+                row.NodeId && host
+                    ? createHref(
+                          routes.node,
+                          {id: row.NodeId, activeTab: 'storage'},
+                          {backend: additionalVersionsProps?.getBackend?.(host)},
```
Comment (artemmufazalov) [inline src/containers/Versions/NodesTable/NodesTable.tsx:46]: Changes from internal version:
- Calculate backend from `additionalVersionsProps.getBackend`

File: src/containers/Versions/NodesTreeTitle/NodesTreeTitle.tsx
Comment (artemmufazalov) [inline src/containers/Versions/NodesTreeTitle/NodesTreeTitle.tsx:1]: Changes from internal version:
- Separate component styles from `GroupedNodesTree`

File: src/containers/Versions/Versions.tsx
Comment (artemmufazalov) [inline src/containers/Versions/Versions.tsx:1]: Changes from internal version:
- Use `GroupByValue` enum
- Update grouping functions params

File: src/utils/versions/getVersionsColors.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    '#ff1493', // deeppink
+    '#00bfff', // deepskyblue
+    '#da70d6', // orchid
+    '#3cb371', // mediumseagreen
+    '#b22222', // firebrick
+];
+
+export const GRAY_COLOR = '#bfbfbf'; // --yc-color-base-neutral-hover
+
+export const getVersionsMap = (versions: string[], initialMap: VersionsMap = new Map()) => {
```
Comment (artemmufazalov) [inline src/utils/versions/getVersionsColors.ts:29]: I use old internal version of `getVersionsMap` and `getVersionToColorMap`, but with updated colours, to get version to color mapping

**Review by Raubzeug — APPROVED** [review]

**Review by Raubzeug — APPROVED** [review]

**Pull request merged**